src/library_html5.js

```diff
@@ -935,17 +935,41 @@ var LibraryHTML5 = {
 
   $fillOrientationChangeEventData__deps: ['$screenOrientation'],
   $fillOrientationChangeEventData: (eventStruct) => {
-    var orientations  = ["portrait-primary", "portrait-secondary", "landscape-primary", "landscape-secondary"];
-    var orientations2 = ["portrait",         "portrait",           "landscape",         "landscape"];
-
-    var orientationString = screenOrientation();
-    var orientation = orientations.indexOf(orientationString);
-    if (orientation == -1) {
-      orientation = orientations2.indexOf(orientationString);
+    // OrientationType enum
+    var orientationsType1 = ['portrait-primary', 'portrait-secondary', 'landscape-primary', 'landscape-secondary'];
+    // alternative selection from OrientationLockType enum
+    var orientationsType2 = ['portrait',         'portrait',           'landscape',         'landscape'];
+
+    var orientationIndex = -1;
+    var orientationAngle =  0;
+    var screenOrientObj  = screenOrientation();
+    if (screenOrientObj) {
+      var orientationType;
+      if (typeof screenOrientObj === 'object') {
+        orientationType  = screenOrientObj.type;
+        orientationAngle = screenOrientObj.angle;
+      } else {
+        // Edge only supports the older string type until 2020 (pre-Chrome)
+        orientationType  = String(screenOrientObj);
+      }
+      orientationIndex = orientationsType1.indexOf(orientationType);
+      if (orientationIndex == -1) {
```

Comment:
Very micro-opt, though in other places I've favored `if (x < 0)` to check for missing index to save a byte.